File: HRMS.Database/Models/TaskComment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HRMS.Database.Models;

[Table("TaskComments")]
public class TaskComment
{
    [Key]
    public int Id { get; set; }

    public DateTime Time { get; set; }

    public string Content { get; set; } = string.Empty;

    [ForeignKey(nameof(Task))]
    public int TaskId { get; set; }
    public virtual Task? Task { get; set; }

    [ForeignKey(nameof(Employee))]
    public int EmployeeId { get; set; }
    public virtual Employee? Employee { get; set; }
}

File: HRMS.Desktop/lib/models/task_comment.dart
import 'package:HRMS/models/task.dart';
import 'package:json_annotation/json_annotation.dart';

import 'employee.dart';

part 'task_comment.g.dart';

@JsonSerializable()
class TaskComment {
  int id;
  DateTime time;
  String content;
  Task? task;
  Employee? employee;

  TaskComment(
    this.id,
    this.time,
    this.content,
    this.task,
    this.employee
  );

  factory TaskComment.fromJson(Map<String, dynamic> json) =>
      _$TaskCommentFromJson(json);

  Map<String, dynamic> toJson() => _$TaskCommentToJson(this);
}
